File: bungee/src/main/java/com/emanuelvini/nuven/core/bungee/data/adapter/ProfileAdapter.java
package com.emanuelvini.nuven.core.bungee.data.adapter;

import com.emanuelvini.nuven.core.bungee.BungeeMain;
import com.emanuelvini.nuven.core.bungee.manager.TagManager;
import com.emanuelvini.nuven.core.shared.player.profile.Profile;
import com.henryfabio.sqlprovider.executor.adapter.SQLResultAdapter;
import com.henryfabio.sqlprovider.executor.result.SimpleResultSet;

public class ProfileAdapter implements SQLResultAdapter<Profile> {

    private final TagManager tagManager = BungeeMain.instance().getTagManager();

    @Override
    public Profile adaptResult(SimpleResultSet simpleResultSet) {
        return new Profile(
                simpleResultSet.get("owner"),
                simpleResultSet.get("cash"),
                tagManager.getTags().get((int) simpleResultSet.get("tag"))
        );
    }
}

File: bungee/src/main/kotlin/com/emanuelvini/nuven/core/bungee/data/adapter/ProfileAdapter.kt
package com.emanuelvini.nuven.core.bungee.data.adapter

import com.emanuelvini.nuven.core.bungee.BungeeMain
import com.emanuelvini.nuven.core.bungee.manager.RoleManager
import com.emanuelvini.nuven.core.shared.player.profile.Profile
import com.henryfabio.sqlprovider.executor.adapter.SQLResultAdapter
import com.henryfabio.sqlprovider.executor.result.SimpleResultSet

class ProfileAdapter : SQLResultAdapter<Profile> {
    private val roleManager: RoleManager = BungeeMain.instance!!.roleManager
    override fun adaptResult(simpleResultSet: SimpleResultSet): Profile {
        val order = simpleResultSet.get("tag") as Int
        print(order)
        print(roleManager.roles.size)
        return Profile(
            simpleResultSet.get("name"),
            simpleResultSet.get("cash"),
            (if (roleManager.roles.size <= order) roleManager.roles[0] else roleManager.roles[order])!!
        )
    }
}
